hermes/hermes.go

```diff
@@ -0,0 +1,95 @@
+	return [...]string{"Create", "Delete"}[fileOperation]
+}
+
+// Intent stores the next intended file operation of Hermes.
+// This is used as part of the StateJournal of Hermes.
+type Intent struct {
+	operation FileOperation `json:"fileoperation"` // Stores the file operation intent, either CREATE or DELETE
+	filename  string        `json:"filename"`      // Stores the filename that the operation is being performed on.
+}
+
+// StateJournal stores the state of Hermes in two parts: the next operation intent and a map of filenames.
+// The intent stores the operation to be performed and the name of the file that the operation is being performed on.
+// The filenames map is a map of file IDs to filenames.
+// If an entry does not exist for a given ID, then the file does not exist (i.e. has been deleted)
+type StateJournal struct {
+	// intent stores the next intended file operation and the name of the file that the operation is being performed on.
+	intent Intent `json:"intent"`
+	// filenames is a map of file IDs to filenames.
+	filenames map[int]string `json:"filenames"`
+}
+
+// Init initialises the map in the StateJournal so that entries can be added to it.
+func (sj *StateJournal) Init() {
+	sj.filenames = make(map[int]string) // initialise filenames to a map
+}
+
+// Hermes is the main Hermes prober that will startup Hermes and initiate monitoring targets.
+type Hermes struct {
+	// Journal stores the state of Hermes as a combination of next operation intent and a filenames map
+	Journal StateJournal
+	// CancelCloudprober is a cancel() function associated with the context passed to Cloudprober when initialised.
+	CancelCloudprober func()
+}
+
+// InitialiseCloudproberFromConfig initialises Cloudprober from the config passed as an argument.
+// Parameters:
+// - config: config should be the contents of a Cloudprober config file. This is most likely: "grpc_port=9314"
+//           -> the "grpc_port:" field is the only required field for the config.
+// Returns:
+// - error:
+//	- logger.NewCloudproberLog() error: error initialising logging on GCE (Stackdriver)
+//	- sysvars.Init():
+//		- error getting local hostname: [error]:
+//			-> error getting hostname from os.Hostname()
+//		- other error
+//			-> error initialising Cloud metadata
+//	- config.ParseTemplate() error:
+//		-> regex compilation issue of config or config could not be processed as a Go text template
+//	- proto.UnmarshalText() error:
+//		-> The config does not match the proto that it is being unmarshalled with.
+//	- initDefaultServer() error:
+//		- failed to parse default port from the env var: [serverEnvVar]=[parsedPort]
+//		- error while creating listener for default HTTP server: [error]
+//	- error while creating listener for default gRPC server: [error]
+//	- tlsconfig.UpdateTLSConfig() error: an error occurred when updating the TLS config from the config passed.
+func (h *Hermes) InitialiseCloudproberFromConfig(config string) error {
```

Comment:
Why is this a method? The receiver (`h`) is not used.
What's the effect of this function? Does it have some side-effect?
If it has nothing to do with the Hermes instance, then remove it and call the function directly from main().

Reply:
That's a good point. This had initially done several other things (such as storing the context in Hermes), but now that it doesn't, it makes sense to remove it.